src/components/modals/ModalWithActionButtons.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { Button, ButtonProps } from '../buttons/Button';
+
+import { BaseModal } from './BaseModal';
+
+export interface ActionButton extends ButtonProps {
+  actionCallback?: (...params: any[]) => any;
```

Comment:
i think it's fine to keep onClick with the (event) => signature, it's pretty standard and so people should expect it.

Reply:
even as I read through this pr i keep forgetting what actionCallback is